plot-lib/plot/private/plot2d/plot-area.rkt

```diff
@@ -226,18 +226,103 @@
          (vector (+ area-x-min (* x area-per-view-x))
                  (- area-y-max (* y area-per-view-y))))))
 
-    (: init-top-margin Real)
-    (define init-top-margin
+
+    (: title-margin Real)
+    (define title-margin
       (let ([title (plot-title)])
         (cond [(and (plot-decorations?) title)
                (if (pict? title)
                    (+ (pict-height title) (* 1/2 char-height))
                    (* 3/2 char-height))]
               [else  0])))
 
+    ;; legend margin calculation and preparation of draw function
+    (: legend-print (-> (Listof legend-entry) Void))
+    (: init-left-margin Real)
+    (: init-right-margin Real)
+    (: init-top-margin Real)
+    (: init-bottom-margin Real)
+    (define-values (legend-print init-left-margin init-right-margin init-top-margin init-bottom-margin)
+      (let* ([legend-anchor (outside-anchor (plot-legend-anchor))]
+             [legend-rect (and legend-anchor
+                               (not (empty? legend))
+                               (send pd calculate-legend-rect
+                                     legend
+                                     (vector (ivl dc-x-min (+ dc-x-min dc-x-size))
+                                             (ivl dc-y-min (+ dc-y-min dc-y-size)))
+                                     legend-anchor))]
+             [gap (pen-gap)]
+             [make-print
+              (λ ([get-bounds : (-> (Listof Real))])
+                (λ ([legend-entries : (Listof legend-entry)])
+                  (match-define (list x-min x-max y-min y-max gap-size) (get-bounds))
+                  (send pd draw-legend legend-entries
+                        (vector (ivl (+ x-min gap-size) (- x-max gap-size))
+                                (ivl (+ y-min gap-size) (- y-max gap-size))))))]
+             [make-none (λ ([get-bounds : (-> (Listof Real))])
+                          (values (make-print get-bounds) 0 0 title-margin 0))]
+             [none (λ () (make-none (λ () (list area-x-min area-x-max area-y-min area-y-max
+                                                (+ gap tick-radius)))))])
+        (cond
+          [legend-rect
+           (define 2*gap (* 2 gap))
+           (define 3*gap (* 3 gap))
```

Comment:
The names `2*gap` and `3*gap` are confusing (I know they are valid, but they can be confusing to anyone who is not familiar with Racket or Scheme).    Can you change them to something that reflects their intent (as oppose to their value).  E.g. they could be called `small-gap` and `large-gap`.